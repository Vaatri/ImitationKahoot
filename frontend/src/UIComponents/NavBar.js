import React from 'react';
import {
  AppBar, Typography, Button, Popper, makeStyles, MenuList, MenuItem, IconButton, Paper,
} from '@material-ui/core';
import MenuIcon from '@material-ui/icons/Menu';
import { useHistory } from 'react-router-dom';

const NavBar = () => {
  const history = useHistory();
  const [menuOpen, setMenuOpen] = React.useState(false);
  const [anchorEl, setAnchorEl] = React.useState(null);
  const toggleMenu = (event) => {
    setAnchorEl(anchorEl ? null : event.currentTarget);
    setMenuOpen(true);
  };

  const open = Boolean(anchorEl);

  const useStyles = makeStyles(() => ({
    navBar: {
      display: 'flex',
      flexDirection: 'row',
    },
  }));

  const handleLogout = () => {
    localStorage.clear();
    history.push('/login');
  };

  const classes = useStyles();

  return (
    <header>
      <AppBar className={classes.navBar} key="nav-bar">
        <IconButton onClick={(event) => toggleMenu(event)}>
          <MenuIcon />
        </IconButton>
        <Popper open={open} anchorEl={anchorEl} disablePortal>
          <Paper>
            <MenuList autoFocusItem={menuOpen}>
              <MenuItem key="1">New Game</MenuItem>
              <MenuItem key="2">Join Game</MenuItem>
            </MenuList>
          </Paper>
        </Popper>
        <Typography variant="h4">BigBrain</Typography>
        <Button onClick={handleLogout}>Logout</Button>
        {/* <Button onClick={(event) => toggleMenu(event)}>clickme</Button> */}
      </AppBar>
    </header>
  );
};

export default NavBar;
