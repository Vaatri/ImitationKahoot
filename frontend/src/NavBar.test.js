import { shallow } from 'enzyme';
import React from 'react';
import NavBar from './UIComponents/NavBar';

describe('NavBar', () => {
  const wrapper = shallow(<NavBar />);

  it('check that the logo has alt tag', () => {
    const image = wrapper.find('img');
    expect(image.prop('alt')).toEqual('BB-logo');
  });

  it('check that the image matches the snapshot', () => {
    const image = wrapper.find('img');
    expect(image).toMatchSnapshot();
  });

  it('check that the create a quiz button has aria-label', () => {
    const button = wrapper.find('#create-quiz');
    expect(button.props()['aria-label']).toEqual('create a quiz');
  });

  it('check that the create a quiz button triggers modal', () => {
    const button = wrapper.find('#create-quiz');
    button.simulate('click');
    expect(button).toMatchSnapshot();
    const modal = wrapper.find('#new-game');
    expect(modal).toMatchSnapshot();
    expect(modal.props()['aria-labelledby']).toEqual('new-game');
  });

  it('check if the join game button text has aria-label', () => {
    const button = wrapper.find('#join');
    expect(button.props()['aria-label']).toEqual('join a game');
  });

  it('check if the join game button text is correct', () => {
    const button = wrapper.find('#join');
    expect(button.text()).toEqual('Join Game');
  });

  it('check if the join game button matches snapshot', () => {
    const button = wrapper.find('#join');
    expect(button).toMatchSnapshot();
  });

  it('check if the logout button text has an aria-label', () => {
    const button = wrapper.find('#logout');
    expect(button.props()['aria-label']).toEqual('logout');
  });

  it('check if the logout button text is correct', () => {
    const button = wrapper.find('#logout');
    expect(button.text()).toEqual('Logout');
  });

  it('check if the logout button matches the snapshot', () => {
    const button = wrapper.find('#logout');
    expect(button).toMatchSnapshot();
  });

  it('check the title of the nav bar is present', () => {
    const title = wrapper.find('#title');
    expect(title.text()).toEqual('BigBrain');
  });

  it('check the title matches the snapshot', () => {
    const title = wrapper.find('#title');
    expect(title).toMatchSnapshot();
  });
});
