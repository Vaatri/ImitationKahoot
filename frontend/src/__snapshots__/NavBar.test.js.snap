// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar check if the join game button matches snapshot 1`] = `ShallowWrapper {}`;

exports[`NavBar check if the join game button text is right and has aria-label 1`] = `ShallowWrapper {}`;

exports[`NavBar check if the logout button matches the snapshot 1`] = `ShallowWrapper {}`;

exports[`NavBar check if the logout button text is right and has an aria-label 1`] = `ShallowWrapper {}`;

exports[`NavBar check that the create a quiz button triggers modal 1`] = `ShallowWrapper {}`;

exports[`NavBar check that the create a quiz button triggers modal 2`] = `ShallowWrapper {}`;

exports[`NavBar check that the create a quiz button triggers modal and has aria-label 1`] = `ShallowWrapper {}`;

exports[`NavBar check that the create a quiz button triggers modal and has aria-label 2`] = `ShallowWrapper {}`;

exports[`NavBar check that the image matches the snapshot 1`] = `ShallowWrapper {}`;

exports[`NavBar check that the logo has alt tag 1`] = `ShallowWrapper {}`;

exports[`NavBar check the title matches the snapshot 1`] = `ShallowWrapper {}`;

exports[`NavBar check the title of the nav bar is present 1`] = `ShallowWrapper {}`;
